### ISSUE TITLE
bug: Android application crash - java.lang.NullPointerException: at com.getcapacitor.Plugin.triggerPermissionCallback (Plugin.java:143)

### ISSUE THREAD

**Issue by TeaKave** [ionic-team/capacitor#4472]
### Capacitor Version

```
Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/cli: 3.0.0-rc.0
  @capacitor/core: 3.0.0-rc.0
  @capacitor/android: 3.0.0-rc.0

```

### Platform(s)

- Android

### Current Behavior

I can not reproduce this crash yet. The application asks for location permission after splash screen.
Android application is crashing with stack trace:
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4962)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:5003)
  at android.app.servertransaction.ActivityResultItem.execute (ActivityResultItem.java:51)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2109)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7682)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:516)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:950)
Caused by: java.lang.NullPointerException: 
  at com.getcapacitor.Plugin.triggerPermissionCallback (Plugin.java:143)
  at com.getcapacitor.Plugin.lambda$initializeActivityLaunchers$1$Plugin (Plugin.java:131)
  at com.getcapacitor.-$$Lambda$Plugin$g-uDx5XqRWyrlYV2wp0-DfAj-NQ.onActivityResult (Unknown Source:6)
  at androidx.activity.result.ActivityResultRegistry.doDispatch (ActivityResultRegistry.java:362)
  at androidx.activity.result.ActivityResultRegistry.dispatchResult (ActivityResultRegistry.java:322)
  at androidx.activity.ComponentActivity.onRequestPermissionsResult (ComponentActivity.java:654)
  at androidx.fragment.app.FragmentActivity.onRequestPermissionsResult (FragmentActivity.java:604)
  at com.getcapacitor.BridgeActivity.onRequestPermissionsResult (BridgeActivity.java:190)
  at android.app.Activity.dispatchRequestPermissionsResult (Activity.java:8289)
  at android.app.Activity.dispatchActivityResult (Activity.java:8139)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4955)
```

### Expected Behavior

Application does not crash.

### Code Reproduction

-

### Other Technical Details

**Package.json:**
```
"dependencies": {
		"@capacitor/android": "^3.0.0-rc.0",
		"@capacitor/app": "^0.3.6",
		"@capacitor/cli": "^3.0.0-rc.0",
		"@capacitor/core": "^3.0.0-rc.0",
		"@capacitor/device": "^0.5.1",
		"@capacitor/geolocation": "^0.4.0",
		"@capacitor/haptics": "^0.3.6",
		"@capacitor/local-notifications": "^0.6.0",
		"@capacitor/push-notifications": "^0.3.6",
		"@capacitor/share": "^0.5.0",
		"@capacitor/splash-screen": "^0.3.6",
		"@capacitor/storage": "^0.3.6",
		"capacitor-rate-app": "^0.1.1",
		"commander": "^7.0.0",
		"cordova-res": "^0.15.3",
		"fs-extra": "^9.1.0",
		"plist": "^3.0.1",
		"zpad": "^0.5.0"
	}
```

**Impacted devices:**

| Android version | Device  | Occurrences | System on chip |
| ------------- | ------------- | ------------- | ------------- |
| Android 10 (SDK 29) | Motorola motorola one 5G UW  | 13  | Qualcomm SM7250  |
| Android 11 (SDK 30) |  Samsung Galaxy S10+  | 5  | Samsung Exynos 9820  |
| Android 10 (SDK 29) |  Huawei P20 Pro  | 3  | HiSilicon KIRIN970  |
| Android 10 (SDK 29), Android 11 (SDK 30) |  Others  | 53  | -  |

`npm --version` output: `7.5.3`

`node --version` output: `v14.15.5`

**Comment by jcesarmobile** [comment]
can you try with rc.1?

**Comment by Ionitron** [comment]
It looks like this issue didn't get the information it needed, so I'll close it for now. If I made a mistake, sorry! I am just a bot.

Have a great day!
Ionitron 💙

**Comment by TeaKave** [comment]
Hi. Thank you for your reply. The issue was present on 3.0.0 but the problem was on our side. We migrated our application from cordova and for some reason the coarse permission was defined in manifest like this: 

```
<uses-permission
        android:name="android.permission.ACCESS_COARSE_LOCATION"
        android:maxSdkVersion="22" />
```
        
So I just removed **maxSdkVersion** for this permission and now everything works fine. 

Sorry for the issue and thank you for your awesome job with capacitor! :)
